web/src/popup_banners.ts

```diff
@@ -21,6 +21,46 @@ const CONNECTION_ERROR_POPUP_BANNER: Banner = {
     custom_classes: "connection-error-banner popup-banner",
 };
 
+// Show user a banner with a button to allow user to navigate
+// to the first unread if required.
+const FOUND_MISSING_UNREADS_IN_CURRENT_NARROW: Banner = {
+    intent: "warning",
+    label: $t({
+        defaultMessage: "This conversation also has older unread messages.",
+    }),
+    buttons: [
+        {
+            attention: "quiet",
+            label: $t({defaultMessage: "Jump to first unread"}),
+            custom_classes: "found-missing-unreads-jump-to-first-unread",
+        },
+    ],
+    close_button: true,
+    custom_classes: "found-missing-unreads popup-banner",
+};
+
+export function open_found_missing_unreads_banner(on_jump_to_first_unread: () => void): void {
+    banners.append(FOUND_MISSING_UNREADS_IN_CURRENT_NARROW, $("#popup_banners_wrapper"));
+
+    $("#popup_banners_wrapper").on(
+        "click",
+        ".found-missing-unreads-jump-to-first-unread",
+        function (this: HTMLElement, e) {
+            e.preventDefault();
+            e.stopPropagation();
+
+            const $banner = $(this).closest(".banner");
+            banners.fade_out_popup_banner($banner);
+            on_jump_to_first_unread();
+        },
+    );
+}
+
+export function close_found_missing_unreads_banner(): void {
+    const $banner = $("#popup_banners_wrapper").find(".found-missing-unreads");
+    banners.fade_out_popup_banner($banner);
+}
```

Comment:
It might be a good follow-up to make this more obviously a `noop` if the banner isn't actually shown.

Reply:
Opened #34033